client/src/components/Profile.jsx

```diff
@@ -8,17 +8,27 @@ class Profile extends Component {
 
     state = {
         user: {},
-        page_id: ""
     };
 
     componentDidMount() {
-        this.setState({ page_id: this.props.match.params.profile_id });
         request('/profile/' + this.props.match.params.profile_id)
-        .then(json => {
-            this.setState({ user: json });
+        .then(data => {
+            this.setState({ user: data });
+            sessionStorage.setItem("User-nickname", this.state.user.nickname);
```

Comment:
This is wrong. What happens if I go to the page of **another** user?
You must checks if I am an owner of this profile.